examples/esp32-with-cred/esp32-server-with-cred.rs

```diff
@@ -190,22 +177,21 @@ mod esp32 {
         info!("starting wifi...");
         #[allow(clippy::redundant_clone)]
         #[cfg(not(feature = "qemu"))]
-        let (ip, _wifi) = {
-            let wifi = start_wifi(
-                periph.modem,
-                sys_loop_stack,
-                &nvs_vars.wifi_ssid,
-                &nvs_vars.wifi_pwd,
-            )
-            .expect("failed to start wifi");
-            (
-                wifi.wifi()
-                    .sta_netif()
-                    .get_ip_info()
-                    .expect("failed to get ip info")
-                    .ip,
-                wifi,
-            )
+        let network = {
+            let wifi = loop {
+                match Esp32WifiNetwork::new(
+                    sys_loop_stack.clone(),
+                    &nvs_vars.wifi_ssid,
+                    &nvs_vars.wifi_pwd,
+                ) {
+                    Ok(wifi) => break wifi,
+                    Err(_) => {
+                        log::info!("wifi could not connect, retrying...");
```

Comment:
This is nice in that now we won't crash if WiFI isn't there. But it is still the case that we can't come up if wifi is down. Should we be able to? We could still, for instance, be collecting sensor data in that state. We'd need a cached config though, which I don't think we have yet. Do we have plans/tickets on how to get there?

Reply:
I was planning on making that adjustment when introducing the data task